mentat/llm_api_handler.py

```diff
@@ -71,9 +74,25 @@ def conversation_tokens(messages: list[ChatCompletionMessageParam], model: str):
         # every message follows <im_start>{role/name}\n{content}<im_end>\n
         num_tokens += 4
         for key, value in message.items():
-            if not isinstance(value, str):
-                continue
-            num_tokens += len(encoding.encode(value))
+            if isinstance(value, list):
+                for entry in value:  # type: ignore
```

Comment:
There are a lot of ignores here. I think Pyright should have enough information given the classes to know entry is of type `ChatCompletionContentPartParam`. Maybe it would help if instead of iterating over items we just accessed the content which should be the only source of tokens anyway. For our use case the only attributes should be role and content. If we start using function calls there could be others.

I'm not sure as written it's exactly correct because when we iterate over items one of them will be the role which should already be counted in line 72. In my `llm_api_handler_test` line 33 I add 5 at the end but I think adding 4 would be correct.

Reply:
It could be a list of ToolCalls (in ChatCompletionAssistantMessageParam). I just added a commit that casts value as long as the key is "content".

 I think the +4 is for the im_start and im_end, not for the role, and the role is always one token which bumps it up to 5. I could be wrong though; I'll try curling some messages to openai (without streaming, so they'll give me the exact tokens) and running them through our function and see if I'm wrong.